src/rsa/blinding.rs

```diff
@@ -0,0 +1,585 @@
+//
+// TODO(djudd) Part of this code can potentially be pulled back into
+// `super::limb` and shared with EC key generation, without unnecessarily
+// complicating that, once specialization is stabilized.
+struct Range<'a> {
+    max_exclusive: &'a [Limb],
+    sampling_params: &'a SamplingParams,
+}
+
+impl <'a> Range<'a> {
+
+    /// Are these little-endian limbs within the range?
+    ///
+    /// Checks in constant time.
+    pub fn are_limbs_within(&self, limbs: &[Limb]) -> bool {
+        assert_eq!(self.max_exclusive.len(), limbs.len());
+
+        let is_zero = limbs_are_zero_constant_time(limbs);
+        let is_lt_max =
+            limbs_less_than_limbs_constant_time(limbs, self.max_exclusive);
+
+        is_zero == LimbMask::False && is_lt_max == LimbMask::True
+    }
+
+    /// Chooses a positive integer within the range and stores it into `dest`.
+    ///
+    /// This function is intended to be suitable for generating private keys.
+    fn sample_into_limbs(&self, dest: &mut [Limb], rng: &rand::SecureRandom)
+                         -> Result<(), error::Unspecified> {
+        // Loosely based on [NSA Suite B Implementer's Guide to ECDSA]
+        // Appendix A.1.2, and
+        // [NSA Suite B Implementer's Guide to NIST SP 800-56A] Appendix B.2,
+        // "Key Pair Generation by Testing Candidates".
+        //
+        // [NSA Suite B Implementer's Guide to ECDSA]: doc/ecdsa.pdf.
+        // [NSA Suite B Implementer's Guide to NIST SP 800-56A]: doc/ecdh.pdf.
+
+        assert_eq!(self.max_exclusive.len(), dest.len());
+
+        // XXX: The value 100 was chosen to match OpenSSL due to uncertainty of
+        // what specific value would be better, but it seems bad to try 100
+        // times.
+        for _ in 0..100 {
+            {
+                let mut dest_as_bytes = limbs_as_bytes_mut(dest);
+                try!(rng.fill(&mut dest_as_bytes));
+            }
+
+            {
+                // Mask off unwanted bits
+                let mask = self.sampling_params.most_sig_limb_mask;
+                dest[self.max_exclusive.len() - 1] &= mask;
+            }
+
+            if self.are_limbs_within(&dest) {
+                return Ok(());
+            }
+
+            if self.sampling_params.reduce_when_over_bound {
+                // `dest` is not in (0, max) but maybe we can fix that.
+                // (See above for explanation of why this is safe.)
+
+                limbs_reduce_once_constant_time(dest, self.max_exclusive);
+                if self.are_limbs_within(&dest) {
+                    // `dest` was in [max, 2*max)
+                    return Ok(());
+                }
+
+                limbs_reduce_once_constant_time(dest, self.max_exclusive);
+                if self.are_limbs_within(&dest) {
+                    // `dest` was in [2*max, 3*max)
+                    return Ok(());
+                }
+
+                // `dest` was in [3*max, 2**(n+1)) or zero
```

Comment:
I suggest that you append ". Continue looping to generate a new random value." Especially because the period is missing, it looks like there might be some code unintentionally missing here.